### PR TITLE
fix codesign parsing error

### PR DESCRIPTION
**Pull request by gdams**
fixes: https://github.com/AdoptOpenJDK/openjdk-build/issues/1502

### CODE REVIEW

**Comment by gdams** [comment]
Test JDK8 Build: https://ci.adoptopenjdk.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk8u/job/jdk8u-linux-x64-hotspot/250/

**Comment by gdams** [comment]
Test JDK11 Build: https://ci.adoptopenjdk.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk11u/job/jdk11u-mac-x64-hotspot/33/

**Comment by gdams** [comment]
all fixed

File: sbin/build.sh [hunk 1]
```diff
@@ -82,7 +82,9 @@ configuringBootJDKConfigureParameter()
 # Configure the boot JDK
 configuringMacOSCodesignParameter()
 {
-  addConfigureArgIfValueIsNotEmpty "--with-macosx-codesign-identity=" "\"${BUILD_CONFIG[MACOSX_CODESIGN_IDENTITY]}\""
+  if [ ! -z "${BUILD_CONFIG[MACOSX_CODESIGN_IDENTITY]}" ]; then
+    addConfigureArg "--with-macosx-codesign-identity=" "\"${BUILD_CONFIG[MACOSX_CODESIGN_IDENTITY]}\""
```
Comment (jerboaa) [inline sbin/build.sh]: It seems strange that this function uses `\"` and thus, cannot use `addConfigureArgIfValueIsNotEmpty`. Perhaps worth explaining in a comment as to why that is?

**Pull request merged**